e2e/scenario/scenario_mariner-gpu.go

```diff
@@ -0,0 +1,33 @@
+package scenario
+
+import (
+	"testing"
+
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+)
+
+// Returns config for the 'gpu' E2E scenario
+func marinergpu() *Scenario {
+	return &Scenario{
+		Name:        "mariner-gpu",
+		Description: "Tests that a GPU-enabled node using a MarinerV2 VHD can be properly bootstrapped",
+		Config: Config{
+			ClusterSelector: NetworkPluginKubenetSelector,
+			ClusterMutator:  NetworkPluginKubenetMutator,
+			BootstrapConfigMutator: func(t *testing.T, nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].VMSize = "Standard_NC6_v3"
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-cblmariner-v2-gen2"
+				nbc.AgentPoolProfile.VMSize = "Standard_NC6_v3"
+				nbc.AgentPoolProfile.Distro = "aks-cblmariner-v2-gen2"
+				nbc.ConfigGPUDriverIfNeeded = true
+				nbc.EnableGPUDevicePluginIfNeeded = false
+				nbc.EnableNvidia = true
+			},
+			VMConfigMutator: func(vmss *armcompute.VirtualMachineScaleSet) {
+				vmss.SKU.Name = to.Ptr("Standard_NC6s_v3")
```

Comment:
you also need to update the image reference of the vmss model so it actually uses a mariner image (Iguess setting distro didn't really matter in this case...?)

you need something similar to: https://github.com/Azure/AgentBaker/blob/master/e2e/scenario/scenario_marinerv2.go#:~:text=vmss.Properties,%7D

Reply:
This also needs to be added for the [ubuntu-gpu scenario](https://github.com/Azure/AgentBaker/blob/978984f63931618f4316176981daab1573fe862f/e2e/scenario/scenario_ubuntu-gpu.go#L28)

Reply:
Fixed

Reply:
> @cameronmeissner Does this also need to be explicitly added for the [ubuntu-gpu scenario](https://github.com/Azure/AgentBaker/blob/978984f63931618f4316176981daab1573fe862f/e2e/scenario/scenario_ubuntu-gpu.go#L28)

nope, since the base vmss model already references and ubuntu 1804 VHD:
https://github.com/Azure/AgentBaker/blob/4e596e6763a0407a839aff08ea357438e2125aad/e2e/vmss.go#:~:text=StorageProfile%3A%20%26,%7D%2C